test/icinga-perfdata.cpp

```diff
@@ -11,309 +11,318 @@ BOOST_AUTO_TEST_SUITE(icinga_perfdata)
 
 	pv = PerfdataValue::Parse("test=1EWh");
 	BOOST_CHECK(pv);
 
-	BOOST_CHECK(pv->GetValue() == 1000.0 * 1000 * 1000 * 1000 * 1000 * 1000);
-	BOOST_CHECK(pv->GetUnit() == "watt-hours");
-	BOOST_CHECK(pv->GetCrit() == Empty);
-	BOOST_CHECK(pv->GetWarn() == Empty);
-	BOOST_CHECK(pv->GetMin() == Empty);
-	BOOST_CHECK(pv->GetMax() == Empty);
+	BOOST_CHECK_EQUAL(pv->GetValue(), 1000.0 * 1000 * 1000 * 1000 * 1000 * 1000);
+	BOOST_CHECK_EQUAL(pv->GetUnit(), "watt-hours");
+	BOOST_CHECK_EQUAL(pv->GetCrit(), Empty);
+	BOOST_CHECK_EQUAL(pv->GetWarn(), Empty);
+	BOOST_CHECK_EQUAL(pv->GetMin(), Empty);
+	BOOST_CHECK_EQUAL(pv->GetMax(), Empty);
 
 	str = pv->Format();
-	BOOST_CHECK(str == "test=1000000000000000000Wh");
+	BOOST_CHECK_EQUAL(str, "test=1000000000000000000Wh");
 
 	pv = PerfdataValue::Parse("test=1000mW");
 	BOOST_CHECK(pv);
 
-	BOOST_CHECK(pv->GetValue() == 1);
-	BOOST_CHECK(pv->GetUnit() == "watts");
-	BOOST_CHECK(pv->GetCrit() == Empty);
-	BOOST_CHECK(pv->GetWarn() == Empty);
-	BOOST_CHECK(pv->GetMin() == Empty);
-	BOOST_CHECK(pv->GetMax() == Empty);
+	BOOST_CHECK_EQUAL(pv->GetValue(), 1);
+	BOOST_CHECK_EQUAL(pv->GetUnit(), "watts");
+	BOOST_CHECK_EQUAL(pv->GetCrit(), Empty);
+	BOOST_CHECK_EQUAL(pv->GetWarn(), Empty);
+	BOOST_CHECK_EQUAL(pv->GetMin(), Empty);
+	BOOST_CHECK_EQUAL(pv->GetMax(), Empty);
 
 	str = pv->Format();
-	BOOST_CHECK(str == "test=1W");
+	BOOST_CHECK_EQUAL(str, "test=1W");
 }
 
 BOOST_AUTO_TEST_CASE(warncritminmax)
 {
 	PerfdataValue::Ptr pv = PerfdataValue::Parse("test=123456B;1000;2000;3000;4000");
 	BOOST_CHECK(pv);
 
-	BOOST_CHECK(pv->GetValue() == 123456);
+	BOOST_CHECK_EQUAL(pv->GetValue(), 123456);
 	BOOST_CHECK(!pv->GetCounter());
-	BOOST_CHECK(pv->GetUnit() == "bytes");
-	BOOST_CHECK(pv->GetWarn() == 1000);
-	BOOST_CHECK(pv->GetCrit() == 2000);
-	BOOST_CHECK(pv->GetMin() == 3000);
-	BOOST_CHECK(pv->GetMax() == 4000);
+	BOOST_CHECK_EQUAL(pv->GetUnit(), "bytes");
+	BOOST_CHECK_EQUAL(pv->GetWarn(), 1000);
+	BOOST_CHECK_EQUAL(pv->GetCrit(), 2000);
+	BOOST_CHECK_EQUAL(pv->GetMin(), 3000);
+	BOOST_CHECK_EQUAL(pv->GetMax(), 4000);
 
-	BOOST_CHECK(pv->Format() == "test=123456B;1000;2000;3000;4000");
+	BOOST_CHECK_EQUAL(pv->Format(), "test=123456B;1000;2000;3000;4000");
 }
 
 BOOST_AUTO_TEST_CASE(ignore_invalid_warn_crit_min_max)
 {
 	PerfdataValue::Ptr pv = PerfdataValue::Parse("test=123456;1000:2000;0:3000;3000;4000");
 	BOOST_CHECK(pv);
-	BOOST_CHECK(pv->GetValue() == 123456);
-	BOOST_CHECK(pv->GetWarn() == Empty);
-	BOOST_CHECK(pv->GetCrit() == Empty);
-	BOOST_CHECK(pv->GetMin() == 3000);
-	BOOST_CHECK(pv->GetMax() == 4000);
+	BOOST_CHECK_EQUAL(pv->GetValue(), 123456);
+	BOOST_CHECK_EQUAL(pv->GetWarn(), Empty);
+	BOOST_CHECK_EQUAL(pv->GetCrit(), Empty);
+	BOOST_CHECK_EQUAL(pv->GetMin(), 3000);
+	BOOST_CHECK_EQUAL(pv->GetMax(), 4000);
+
+	BOOST_CHECK_EQUAL(pv->Format(), "test=123456;;;3000;4000");
+}
 
-	BOOST_CHECK(pv->Format() == "test=123456");
+BOOST_AUTO_TEST_CASE(empty_warn_crit_min_max)
```

Comment:
I missed pointing this out to you yesterday, but you need to add this new case [here](https://github.com/Icinga/icinga2/blob/master/test/CMakeLists.txt#L49), otherwise `make test` won't run it.

Reply:
Forgot to push the changes that address this one?

Reply:
It was a bit messy when splitting into two commits. Should be ok now.